### PR TITLE
A quick rundown of codebase structure.

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -75,6 +75,69 @@ You do not need to repeatedly install after modifying python files.
 In case you want to reinstall, make sure that you uninstall pytorch first by running `pip uninstall torch`
 and `python setup.py clean`. Then you can install in `build develop` mode again.
 
+## Codebase structure
+
+* [c10](c10) - Core library files that work everywhere (server, mobile)
```
Comment (resistor) [inline CONTRIBUTING.md]: Maybe want to say something about aten -> c10 here?

**Comment by resistor** [comment]
I love it!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -75,6 +75,69 @@ You do not need to repeatedly install after modifying python files.
...
+    * [api](torch/csrc/api) - The PyTorch C++ frontend.
+    * [distributed](torch/csrc/distributed) - Distributed training
+      support for PyTorch.
+* [tools](tools) - Code generation scripts for the PyTorch library
+  * [autograd](tools/autograd) - Code generation for autograd.  This
+    includes definitions of all our derivatives.
+  * [jit](tools/jit) - Code generation for JIT
+  * [amd_build](tools/amd_build) - HIPify scripts, for transpiling CUDA
+    into AMD HIP.
+* [test](tests) - Python unit tests for PyTorch
```
Comment (resistor) [inline CONTRIBUTING.md]: Another spot that would be good to flesh out.

**Review by resistor — APPROVED** [review]

**Comment by jeffreyksmithjr** [comment]
👏 @ezyang . Great stuff!

**Pull request closed without merging**